main.go

```diff
@@ -126,73 +109,251 @@ func (e *periodicExporter) Collect(ch chan<- prometheus.Metric) {
+	e.counter(metricsChan, labels, labelValues, "mesos_master_tasks_finished_total", float64(stats.TasksFinished), "Total of finished tasks")
+	e.counter(metricsChan, labels, labelValues, "mesos_master_tasks_killed_total", float64(stats.TasksKilled), "Total of killed tasks")
+	e.counter(metricsChan, labels, labelValues, "mesos_master_tasks_lost_total", float64(stats.TasksLost), "Total of lost tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_master_tasks_running", float64(stats.TasksRunning), "Amount of running tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_master_tasks_staging", float64(stats.TasksStaging), "Amount of staging tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_master_tasks_starting", float64(stats.TasksStarting), "Amount of starting tasks")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_master_uptime_seconds", stats.UptimeSecs, "Master uptime in seconds")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_master_valid_framework_to_executor_messages_total", float64(stats.ValidFrameworkToExecutorMessages), "Total of valid framework-to-executor messages")
+	e.counter(metricsChan, labels, labelValues, "mesos_master_valid_status_update_acknowledgements_total", float64(stats.ValidStatusUpdateAcknowledgements), "Total of valid status update acknowledgements")
+	e.counter(metricsChan, labels, labelValues, "mesos_master_valid_status_updates_total", float64(stats.ValidStatusUpdates), "Total of valid status updates")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_master_registrar_queued_operations_total", float64(stats.RegistrarQueuedOperations), "Total of queued operations in registrar")
+	e.gauge(metricsChan, labels, labelValues, "mesos_master_registrar_registry_size_bytes", float64(stats.RegistrarSizeBytes), "Registrar registry size in bytes")
+	e.gauge(metricsChan, labels, labelValues, "mesos_master_registrar_state_fetch_duration_milliseconds", stats.RegistrarStateFetchMilliSecs, "Duration of state fetch by registrar (in milliseconds)")
+	e.counter(metricsChan, labels, labelValues, "mesos_master_registrar_state_store_total", float64(stats.RegistrarStateStoreMilliSecsCount), "Total of state stores by registrar")
+	e.gauge(metricsChan, labels, labelValues, "mesos_master_registrar_state_store_duration_milliseconds_max", stats.RegistrarStateStoreMilliSecsMax, "Maximum duration of state store by registrar (in milliseconds)")
+	e.gauge(metricsChan, labels, labelValues, "mesos_master_registrar_state_store_duration_milliseconds_min", stats.RegistrarStateStoreMilliSecsMin, "Minimum duration of state store by registrar (in milliseconds)")
+	summaryQuantileLabels := append(labels, "quantile")
+	e.gauge(metricsChan, summaryQuantileLabels, append(labelValues, "0.5"), "mesos_master_registrar_state_store_duration_milliseconds", stats.RegistrarStateStoreMilliSecsP50, "Duration of state store by registrar (in milliseconds)")
```

Comment:
Over what time period is this quantile?

If you've going to export quantiles you should put them in a Summary rather than using Gauges. I'd suggest dropping the quantiles.